src/main/java/com/gooddata/dataload/processes/ProcessService.java

```diff
@@ -95,6 +97,33 @@ public DataloadProcess createProcess(Project project, DataloadProcess process) {
     }
 
     /**
+     * Create new ruby process from ruby appstore.
+     * Processes can be deployed only from predefined repos, which are substituted by placeholders
+     *
+     * @param project project to which the process belongs
+     * @param process to create
+     * @param repoPlaceholderName name of repo placeholder, (e.g. PUBLIC_APPSTORE)
+     * @param commitIdentifier string in format commit/COMMIT_HASH or tag/TAG or branch/BRANCH
+     * @param pathInRepo string representing path in git repo to application (e.g. /test/rubyHello)
+     * @return created process
+     */
+    public FutureResult<DataloadProcess> createProcess(Project project, DataloadProcess process, String repoPlaceholderName, String commitIdentifier,
+                                                       String pathInRepo) {
+        notNull(project, "project");
+        notNull(process, "process");
+        notNull(repoPlaceholderName, "placeholderName");
+        notNull(commitIdentifier, "commitIdentifier");
+        notNull(pathInRepo, "pathInRepo");
+        isTrue(process.getType().equals(ProcessType.RUBY.toString()), "only ruby processes can be deployed from appstore");
```

Comment:
IMHO: it is questionable whether this should be checked by SDK - what if GoodData allows more process types in the future?